src/main/java/build/buildfarm/worker/InputFetcher.java

```diff
@@ -194,7 +196,7 @@ private long fetchPolled(Stopwatch stopwatch) throws InterruptedException {
               queuedOperation.getCommand());
     } catch (IOException e) {
       log.log(Level.SEVERE, format("error creating exec dir for %s", operationName), e);
-      owner.error().put(operationContext);
+      failOperation("Error creating exec dir", e.toString());
```

Comment:
Without the delivery to the error() path, there may be existing behavior that we're losing here. The call to failOperation should decorate the context and notify the error() path, where it should perform the putOperation below in failOperation.

Reply:
failedOperation updates the context and forwards it to `owner.error().put(operationContext);` as before.